### PR TITLE
Show proposal time-stamp instead of block counts

### PR DESCRIPTION
**Pull request by cis-shubham-t**
Task: https://particl.atlassian.net/projects/PD/issues/PD-459

### CODE REVIEW

**Comment by cis-shubham-t** [comment]
testing needed!!

File: src/app/wallet/proposals/models/proposal.model.ts [hunk 1]
```diff
@@ -21,8 +24,17 @@ export class Proposal {
     this.type = object.type;
     this.hash = object.hash;
     this.id = object.id;
+    this.createdAt = object.createdAt;
+    this.expiredAt = object.expiredAt;
   }
 
+  getTimeStamp(type: string): String {
```
Comment (zaSmilingIdiot) [inline src/app/wallet/proposals/models/proposal.model.ts]: The type parameter here is unnecessary - the condition below can be based on whether `this.expiredAt > Date.now()`

File: src/app/wallet/proposals/proposal-details/proposal-details.component.html [hunk 2]
```diff
@@ -13,17 +13,17 @@
     <!-- TODO: only show "blocks-left" on ACTIVE proposals -->
     <div class="blocks-left" *ngIf="selectedTab == 'active'">
       <mat-progress-bar mode="determinate" value="{{proposal?.votingProgressPercentage(currentBlockCount)}}" matTooltip="Blocks left for voting" matTooltipPosition="before"></mat-progress-bar>
-      <div class="text">{{proposal?.leftVotingEndBlockCount(currentBlockCount)}} blocks left</div>
+      <div class="text">{{proposal?.getTimeStamp('left')}} time left</div>
```
Comment (zaSmilingIdiot) [inline src/app/wallet/proposals/proposal-details/proposal-details.component.html]: No need for the parameter

File: src/app/wallet/proposals/proposal-details/proposal-details.component.html [hunk 3]
```diff
@@ -13,17 +13,17 @@
...
     </div>
     <!-- TODO: only show "blocks-left" on PLANNED/future proposals -->
     <div class="blocks-left" *ngIf="selectedTab == 'planned'">
       <mat-progress-bar mode="determinate" value="0" matTooltip="Blocks left until voting starts" matTooltipPosition="before"></mat-progress-bar>
-      <div class="text">{{proposal?.leftVotingEndBlockCount(currentBlockCount)}} blocks left until voting starts</div>
+      <div class="text">{{proposal?.getTimeStamp('left')}} time left until voting starts</div>
```
Comment (zaSmilingIdiot) [inline src/app/wallet/proposals/proposal-details/proposal-details.component.html]: No need for the parameter

File: src/app/wallet/proposals/proposal-details/proposal-details.component.html [hunk 4]
```diff
@@ -13,17 +13,17 @@
...
     </div>
     <!-- TODO: only show "blocks-left.results" on PAST proposals -->
     <div class="blocks-left results" *ngIf="selectedTab == 'past'">
       <mat-progress-bar mode="determinate" value="100"></mat-progress-bar>
-      <div class="text">voting finished on block {{proposal?.blockEnd}}</div>
+      <div class="text">voting finished {{proposal?.getTimeStamp('ago')}} ago</div>
```
Comment (zaSmilingIdiot) [inline src/app/wallet/proposals/proposal-details/proposal-details.component.html]: No need for the parameter

**Review by zaSmilingIdiot — CHANGES_REQUESTED** [review]

**Review by zaSmilingIdiot — APPROVED** [review]

**Pull request merged**